### ISSUE TITLE
NPM ERR! Cannot read property 'length' of undefined

### ISSUE THREAD

**Issue by joshua-mbg** [termux/termux-packages#2737]
npm/npm#19265
npm/npm#1855

This disruptive bug is still not elegantly fixed. Can we get a reliable fix from standard install please?

```
Unpacking nodejs-current (9.11.1) ...
Processing triggers for man (1.14.4) ...
Setting up nodejs-current (9.11.1) ...
$ node -v
v9.11.1
$ npm -v
5.6.0
```

**Comment by laem** [comment]
Would it be resolved by updating nodejs-current to v10 #2621 ?

**Comment by F0RIS** [comment]
I have  same error :/

**Comment by TheLastDestroyer** [comment]
replace $cpu with number of cores (or 1 if unsure)

sed -i "s|require('os').cpus().length|$cpu|g" /data/data/com.termux/files/usr/lib/node_modules/npm/node_modules/worker-farm/lib/farm.js

**Comment by fornwall** [comment]
See https://github.com/termux/termux-packages/issues/1798

**Comment by hltdev8642** [comment]
It should be noted that it may have to do with it being the beta version of termux because I couldn't get this fixed to work but as soon as I downgraded to the non beta version of termux and applied this fix it worked right away

**Comment by ctp-placebo** [comment]
> replace $cpu with number of cores (or 1 if unsure)
> 
> sed -i "s|require('os').cpus().length|$cpu|g" /data/data/com.termux/files/usr/lib/node_modules/npm/node_modules/worker-farm/lib/farm.js

Can someone clarify? I have seen this 7 ways on seven sites/threads.   
What i have in the un-edited file:
`, maxConcurrentWorkers : (require('os').cpus() ||  { length: 1 }).length`   

I have tried changing it to a hardcoded 1, 4, 8 (8 core phone) and deleting the rest of the line.   
I have tried changing the `{ length: 1 }` to `{ length: 8 }` and to 4, and even tried reversing the order of the statement so the OR above reads first (long-shot). 
I tried apt install yarn and then yarn add npm.  Nothing makes npm work. 

Using 
nodejs 11.6.0  
npm 6.5.0
Android 9 (OxygenOS 9.0.2)
Termux & apt fully updated system.  

Is there a solution to this?!
